src/praisonai/praisonai/cli/features/custom_definitions.py

```
@@ -461,6 +504,55 @@ def _escape_shell_substitution(text: str) -> str:
+
+        Each command runs in ``working_dir`` (defaulting to the current
+        directory) with a timeout and a captured-output byte cap. Failures are
+        surfaced as :class:`ShellSubstitutionError` so a broken command does not
+        silently produce an empty prompt.
+        """
+        cwd = str(working_dir) if working_dir else None
+
+        def replace_command(match):
+            command = match.group(1).strip()
+            try:
+                completed = subprocess.run(
```

Ruff (0.15.18)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)